src/core/Akka.FSharp/FsApi.fs

```diff
@@ -374,9 +379,12 @@ module Linq =
 
     let toExpression<'Actor>(f : System.Linq.Expressions.Expression) = 
             match f with
-            | Lambda(_, (Call(null, Method "ToFSharpFunc", Ar [| Lambda(_, p) |]))) 
-            | Call(null, Method "ToFSharpFunc", Ar [| Lambda(_, p) |]) -> 
-                Expression.Lambda(p, [||]) :?> System.Linq.Expressions.Expression<System.Func<'Actor>>
+            | Lambda(_, body) -> 
+                match body with  
+                | :? ConstantExpression as cst -> Some(cst.Value, cst.Type)
+                | _ -> failwith "Doesn't match"
+            //| Call(null, Method "ToFSharpFunc", Ar [| Lambda(_, p) |]) -> 
```

Comment:
@Aaronontheweb am still pondering if this will have some side effects. Right now are tests are passing!

Reply:
This might just be bike shedding or nitpicking but I wonder if the result type might be a good fit here. That way you can get your fail but not have to worry about consumers forgetting to catch.

Reply:
What would that look like? My F# experience is limited to managing FAKE scripts

Reply:
Result type is a lot like Option, but with an Ok<T> | Error<U> instead of a Some<T> | None. You can pass along a message about what went wrong in the err, or what specifically didn't match and the consumer is forced to cover the possible failure. It communicates to the consumer that "Failure is something that can happen, and may happen and you need to have a response prepared to it". Now some people will match on the Error and just failwith as you did here, but they must make that decision.

Reply:
For an example... 
```fsharp
    match body with  
    | :? ConstantExpression as cst -> Ok(cst.Value, cst.Type)
    | _ -> Error (sprintf  "Unable to identify the expression %s" typeof<body>)
```

you could also just put 
```fsharp
    match body with  
    | :? ConstantExpression as cst -> Ok(cst.Value, cst.Type)
    | _ -> Error "Doesn't match"`
```

The consumer basically consumes it the same way they would Some | None, but they use 
```fsharp
match result with
| Ok(val,typename) -> failwith "not implemented"
| Error message -> failwith "not implemented"
```

Appologies if there's any slop there, I'm in a bit of a hurry lol. I'll be back later.

Reply:
@eaba think you can do this? https://twitter.com/7sharp9_/status/1344010659021922306